social_core/utils.py

```diff
@@ -111,7 +115,7 @@ def sanitize_redirect(hosts, redirect_to):
 
 def user_is_authenticated(user):
     if user and hasattr(user, 'is_authenticated'):
-        if isinstance(user.is_authenticated, collections.Callable):
+        if isinstance(user.is_authenticated, Callable):
```

Comment:
You can just use built-in function `callable()`.
```suggestion
        if callable(user.is_authenticated):
```